jtelegrambotapi-core/src/main/java/com/jtelegram/api/commands/CommandRegistry.java

```diff
@@ -1,60 +1,69 @@
 package com.jtelegram.api.commands;
 
 import com.jtelegram.api.TelegramBot;
-import com.jtelegram.api.commands.listeners.CommandListener;
-import com.jtelegram.api.commands.listeners.TextCommandListener;
+import com.jtelegram.api.commands.filters.CommandFilter;
+import com.jtelegram.api.commands.filters.RootFilter;
+import com.jtelegram.api.commands.filters.TextFilter;
 import com.jtelegram.api.events.EventHandler;
 import com.jtelegram.api.events.message.TextMessageEvent;
 import com.jtelegram.api.message.entity.MessageEntity;
 import com.jtelegram.api.message.entity.MessageEntityType;
 import com.jtelegram.api.message.impl.TextMessage;
 
 import java.util.*;
+import java.util.stream.Collectors;
 
 public class CommandRegistry implements EventHandler<TextMessageEvent> {
     private final TelegramBot bot;
-    private final Map<CommandListener, CommandHandler> listeners = new HashMap<>();
+    private final List<CommandFilter> listeners = new ArrayList<>();
 
     public CommandRegistry(TelegramBot bot) {
         this.bot = bot;
         bot.getEventRegistry().registerEvent(TextMessageEvent.class, this);
     }
 
-    public void registerCommand(CommandListener listener, CommandHandler handler) {
-        this.listeners.put(listener, handler);
+    public void registerCommand(CommandFilter filter, CommandFilter... filters) {
+        this.listeners.add(new RootFilter(filter, filters));
     }
 
-    public void registerCommand(String command, CommandHandler handler) {
-        this.registerCommand(new TextCommandListener(command, false), handler);
+    public void registerCommand(String command, CommandFilter filter, CommandFilter... filters) {
```

Comment:
A) You cannot use a lambda for this (You have to instantiate CommandHandler which is a filter for some reason)
B) This overall design isn't that intuitive

We need to have this work by having a basis of a command (basic string expression) and a consumer, from there further filters like ChatType, argument length, or whatever can be put on top as additions. Having the handler be a "filter" is confusing and is what makes this new system difficult to use.

Reply:
Lambdas are fixed :)
The signature is `boolean test(TextMessageEvent, Command)`.
Or cast to `CommandHandler` and enjoy `void onCommand(TextMessageEvent, Command)`